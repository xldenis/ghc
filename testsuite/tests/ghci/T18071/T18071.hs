{-# LANGUAGE FlexibleInstances #-}
{-# LANGUAGE PolyKinds #-}
{-# LANGUAGE QuantifiedConstraints #-}
{-# LANGUAGE UndecidableInstances #-}
{-# LANGUAGE FlexibleContexts #-}
module Bug where

import Data.Kind
import Data.Proxy

class    (forall (z :: k). Show (Proxy z)) => ShowProxy (k :: Type) where
instance (forall (z :: k). Show (Proxy z)) => ShowProxy (k :: Type)

data T :: Type -> Type

data U = A | B
