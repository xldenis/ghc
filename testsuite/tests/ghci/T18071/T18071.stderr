
T18071.hs:11:1: warning: [-Wsimplifiable-class-constraints (in -Wdefault)]
    • The constraint ‘Show (Proxy z)’ matches
        instance forall k (s :: k). Show (Proxy s)
          -- Defined in ‘Data.Proxy’
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the quantified constraint ‘forall (z :: k). Show (Proxy z)’
      In the context: forall (z :: k). Show (Proxy z)
      While checking the super-classes of class ‘ShowProxy’
      In the class declaration for ‘ShowProxy’

T18071.hs:12:10: warning: [-Wsimplifiable-class-constraints (in -Wdefault)]
    • The constraint ‘Show (Proxy z)’ matches
        instance forall k (s :: k). Show (Proxy s)
          -- Defined in ‘Data.Proxy’
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the quantified constraint ‘forall (z :: k). Show (Proxy z)’
      In the instance declaration for ‘ShowProxy (k :: Type)’
